src/lib/storage/constraints/table_key_constraint.hpp

```diff
@@ -21,11 +21,22 @@ class TableKeyConstraint final : public AbstractTableConstraint {
    * voilating the set semantics of the constraint.
    */
   TableKeyConstraint(const std::set<ColumnID>& columns, const KeyConstraintType key_type);
+  TableKeyConstraint(const std::set<ColumnID>& columns, const KeyConstraintType key_type, CommitID last_validated_on);
```

Comment:
Because of the datatype there is no reason for it to be const

Reply:
I believe one could make last_validated_on `const, because we don't want to change the method argument value in the constructor (or setter) and const kind of just promises that we will not change the method argument value. The datatype shouldn't have an impact on making this variable const.